qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/settings/QuteGeneralClientSettings.java

```diff
@@ -80,37 +100,56 @@ public static QuteGeneralClientSettings getGeneralQuteSettings(Object initializa
 
 		private final boolean validationSettingsChanged;
 
-		public SettingsUpdateState(boolean validationChanged) {
+		private final boolean codeLensSettingsChanged;
+
+		public SettingsUpdateState(boolean validationChanged, boolean codeLensChanged) {
 			this.validationSettingsChanged = validationChanged;
+			this.codeLensSettingsChanged = codeLensChanged;
 		}
 
 		/**
 		 * Returns true if validation settings changed and false otherwise.
-		 * 
+		 *
 		 * @return true if validation settings changed and false otherwise.
 		 */
 		public boolean isValidationSettingsChanged() {
 			return validationSettingsChanged;
 		}
 
+		/**
+		 * Returns true if CodeLens settings changed and false otherwise.
+		 *
+		 * @return true if CodeLens settings changed and false otherwise.
+		 */
+		public boolean isCodeLensSettingsChanged() {
+			return codeLensSettingsChanged;
+		}
+
 	}
 
 	/**
 	 * Update the given shared settings with the given client settings.
-	 * 
+	 *
 	 * @param sharedSettings the shared settings to update.
 	 * @param clientSettings the client settings used to update the shared settings.
-	 * 
+	 *
 	 * @return the state of the update of the shared settings.
 	 */
 	public static SettingsUpdateState update(SharedSettings sharedSettings, QuteGeneralClientSettings clientSettings) {
 		Map<String, QuteGeneralClientSettings> workspaceFolders = clientSettings.getWorkspaceFolders();
+

```

Comment:
Please do like https://github.com/redhat-developer/quarkus-ls/pull/593/files#diff-397d8fbc3f031e9a83f1e1b098bbe458d76a776cfc8412f537abef5dc87202ceR153 to call cleanWorkspaceFolderSettings just one time.